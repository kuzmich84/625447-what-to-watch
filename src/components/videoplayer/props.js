import PropTypes from "prop-types";

export default {
  handlerMouseOverCard: PropTypes.func.isRequired,
  handlerMouseOutCard: PropTypes.func.isRequired,
  film: PropTypes.object.isRequired,
};

