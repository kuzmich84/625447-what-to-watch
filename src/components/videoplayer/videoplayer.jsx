import React, {PureComponent} from "react";
import {Link} from "react-router-dom";
import propsVideoPlayer from './props';


export default class VideoPlayer extends PureComponent {
  constructor(props) {
    super(props);

    this.videoRef = React.createRef();
    this.start = null;

    this._startVideo = this._startVideo.bind(this);
    this._stopVideo = this._stopVideo.bind(this);
  }

  _startVideo() {
    const video = this.videoRef.current;
    this.start = setTimeout(() => video.play(), 1000);
  }

  _stopVideo() {
    clearTimeout(this.start);
  }

  componentDidUpdate() {
    const video = this.videoRef.current;
    if (this.props.isPlaying) {
      this._startVideo();
    } else {
      this._stopVideo();
      video.pause();
      video.currentTime = 0;
      video.load();
    }
  }

  componentWillUnmount() {
    this._stopVideo();
  }

  render() {
    const {handlerMouseOverCard, handlerMouseOutCard, film, i} = this.props;
    const {id, name, previewImage, videoLink} = film;

    return (
      <article
        className="small-movie-card catalog__movies-card"
        onMouseOver={() => handlerMouseOverCard(film, i)}
        onMouseOut={() => handlerMouseOutCard()}
      >
        <div className="small-movie-card__image">
          <video ref={this.videoRef} width="280" height="175" muted poster={previewImage}>
            <source src={videoLink} type='video/mp4'/>
          </video>
        </div>
        <h3 className="small-movie-card__title">
          <Link className="small-movie-card__link" to={`/films/${id}`}>{name}</Link>
        </h3>
      </article>

    );
  }
}

VideoPlayer.propTypes = propsVideoPlayer;


