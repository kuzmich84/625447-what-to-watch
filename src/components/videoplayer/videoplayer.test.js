import React from "react";
import renderer from "react-test-renderer";
import {VideoPlayer} from "./videoplayer";
import mockFilList from "../../mocks/films";
import {BrowserRouter as Router} from 'react-router-dom';

const noop = () => {};

test(`Should VideoPlayer component render correctly`, () => {
  const tree = renderer
    .create(<Router><VideoPlayer
      handlerMouseOverCard={noop}
      film={mockFilList[2]}
      handlerMouseOutCard={noop}
    /></Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

