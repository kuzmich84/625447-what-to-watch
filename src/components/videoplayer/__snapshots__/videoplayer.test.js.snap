// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should VideoPlayer component render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      muted={true}
      width="280"
    >
      <source
        type="video/mp4"
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/3"
      onClick={[Function]}
    >
      Aviator
       
    </a>
  </h3>
</article>
`;
